earthquakes/notebooks/source memory.ipynb

```diff
@@ -0,0 +1,402 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Using Data From Hard-Disck Memory\n",
+    "Here some functions that uses data from memory to create the ML model\n",
+    "\n",
+    "#### TODO:\n",
+    "- try parallel computing"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#  Data Manipulation\n",
+    "import os\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "pd.options.display.precision = 15\n",
+    "np.seterr(divide='ignore', invalid='ignore')\n",
+    "\n",
+    "#  Machine learning\n",
+    "from xgboost import XGBRegressor\n",
+    "from sklearn.ensemble import GradientBoostingRegressor, RandomForestRegressor, ExtraTreesRegressor\n",
+    "from sklearn.linear_model import LinearRegression\n",
+    "from sklearn.model_selection import train_test_split, cross_val_score\n",
+    "\n",
+    "# Other libraries\n",
+    "from tqdm import tqdm\n",
+    "from os import listdir\n",
+    "from os.path import isfile, join\n",
+    "\n",
+    "#  Our functions\n",
+    "import sys\n",
+    "sys.path.append(\"../..\")\n",
+    "from earthquakes.helpers import Caller, create_feature_dataset_source, create_signal_dataset\n",
+    "from earthquakes.engineering import FeatureComputer\n",
+    "from earthquakes.modeling import predict_on_test\n",
+    "from common.utils import save_object, load_object\n",
+    "\n",
+    "#  Data directories\n",
+    "data_dir = \"../data\"\n",
+    "save_dir = \"../data/chunk_signal\"\n",
+    "save_class = \"../data/classes\"\n",
+    "data_cycle = \"../data/data cycles\"\n",
+    "data_dp = \"../data/data deep learning\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 0. Set up the data\n",
+    "First, the data must be stored in memory in chunks. In order to do so, __we have to run this block of code only ones__. Unless you want to change parameters such as size. In this case, you have to run it again. Be aware that you have to delete the files on the destiny folder."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# train = pd.read_csv(os.path.join(data_dir, \"train.csv\"),\n",
+    "#                     dtype={\"acoustic_data\": np.int16, \"time_to_failure\": np.float64})\n",
+    "train = pd.read_pickle(os.path.join(data_dir, \"train.pickle\"))\n",
+    "caller_cls = Caller(save_dir=save_dir, size=150000)\n",
+    "caller_cls.save_data(train)\n",
+    "save_object(save_path=os.path.join(save_class, \"caller_cls.pkl\"), object_ = caller_cls)\n",
```

Comment:
I understand that creating folders automatically can be convenient, but also risky because the user can lose track of where the folders are. I would prefer to keep the automatic folder creation at a minimum as possible.